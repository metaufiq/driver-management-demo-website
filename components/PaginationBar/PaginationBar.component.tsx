import { FiChevronLeft, FiChevronRight } from "react-icons/fi";

import { Component } from "../../index.types";
import { NavigationButton } from "../Button";
import { PaginationContainer } from "./PaginationBar.component.styles";
import { Props } from "./PaginationBar.component.types";

const MINIMUM_PAGE_INDEX = 1;

const PaginationBar:Component<Props> = ({onNext, onPrev, pageIndex, totalPages})=>(
  <PaginationContainer>
    <NavigationButton 
      onClick={onPrev} 
      disabled={pageIndex===MINIMUM_PAGE_INDEX}
    >
      <FiChevronLeft/>
      Prev Page
    
    </NavigationButton>
    <NavigationButton 
      onClick={onNext} 
      disabled={pageIndex===totalPages}
    >
      Next Page
      <FiChevronRight/>
    </NavigationButton>
</PaginationContainer>
) 

export default PaginationBar