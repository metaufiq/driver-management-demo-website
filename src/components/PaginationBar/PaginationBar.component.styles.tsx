import styled from "styled-components";

export const PaginationContainer = styled.div`
  display: flex;
  column-gap: 1rem;
  justify-content: center;
`