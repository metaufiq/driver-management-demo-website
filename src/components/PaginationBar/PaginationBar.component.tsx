import { ReactElement } from "react";
import { FiChevronLeft, FiChevronRight } from "react-icons/fi";

import { Component } from "../../../index.types";
import Button from "../Button";
import { PaginationContainer } from "./PaginationBar.component.styles";
import { Props } from "./PaginationBar.component.types";

const MINIMUM_PAGE_INDEX = 1;

/**
 * Pagination Bar components
 * @param {Props} props component props 
 * @returns {ReactElement} Pagination Bar components
 */
const PaginationBar:Component<Props> = (
  {onNext, onPrev, pageIndex, totalPages}: Props
): ReactElement=>(
  <PaginationContainer>
    <Button
      onClick={onPrev} 
      disabled={pageIndex===MINIMUM_PAGE_INDEX}
      buttonType='NAVIGATION'
    >
      <FiChevronLeft/>
      Prev Page
    
    </Button>
    <Button 
      onClick={onNext} 
      disabled={pageIndex===totalPages}
      buttonType='NAVIGATION'
    >
      Next Page
      <FiChevronRight/>
    </Button>
</PaginationContainer>
) 

export default PaginationBar