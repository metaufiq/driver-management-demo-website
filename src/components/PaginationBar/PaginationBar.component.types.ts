export type Props = {
  pageIndex: number,
  totalPages: number,
  onPrev: VoidFunction,
  onNext: VoidFunction,
}