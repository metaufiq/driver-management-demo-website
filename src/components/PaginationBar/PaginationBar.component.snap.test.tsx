import { assertSnapshots } from "../../test";
import PaginationBar from ".";
import { Configs } from "../../test/index.types";
import { Props } from "./PaginationBar.component.types";

describe('PaginationBarSnapshot', () => {

  const configs: Configs<Props> = [
    {
      desc: 'should render normal pagination bar',
      props:{
        onNext: ()=>{},
        onPrev: ()=>{},
        pageIndex: 1,
        totalPages: 5
      }
    }
  ];

  assertSnapshots(PaginationBar, configs);
});
