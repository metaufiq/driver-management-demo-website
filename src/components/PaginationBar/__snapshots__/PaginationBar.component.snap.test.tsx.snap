// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBarSnapshot should render normal pagination bar 1`] = `
<div
  className="sc-gsnTZi kzffjj"
>
  <button
    className="sc-bczRLJ hQcwZr"
    disabled={true}
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="15 18 9 12 15 6"
      />
    </svg>
    Prev Page
  </button>
  <button
    className="sc-bczRLJ hQcwZr"
    disabled={false}
    onClick={[Function]}
  >
    Next Page
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="9 18 15 12 9 6"
      />
    </svg>
  </button>
</div>
`;
